src/stormpath.social-login.facebook.js

```diff
@@ -0,0 +1,67 @@
+(function() {
+  'use strict';
+
+  function loginCallback(deferred, response) {
+    var data;
+
+    switch (response.status) {
+      case 'connected':
+        deferred.resolve({
+          providerData: {
+            providerId: 'facebook',
+            accessToken: response.authResponse.accessToken
+          }
+        });
+        break;
+
+      case 'not_authorized':
+        deferred.reject(new Error('Please log into this app'));
+        break;
+
+      default:
+        deferred.reject(new Error('Please log into Facebook.'));
+    }
+  }
+
+  function FacebookLoginService($window, $q, $spJsLoader) {
```

Comment:
Instead of declaring it as one function that returns an object, I would add the init and login methods as prototypes of FacebookLoginService. I.e:

```js
function FacebookLoginService($window, $q, $spJsLoader) {
  this.name = 'Facebook';
  this.clientId = null;
}

FacebookLoginService.prototype.init = function init() {
  // ...
};

FacebookLoginService.prototype.login = function login(options) {
  // ...
};
```